src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,27 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
+            return null;
+        }
+
+        public override BoundNode VisitThrowStatement(BoundThrowStatement node)
+        {
+            VisitThrow(node.ExpressionOpt);
+            return null;
+        }
+
+        private void VisitThrow(BoundExpression expr)
+        {
+            var result = VisitRvalueWithResult(expr);
+            if ((expr?.ConstantValue?.IsNull == true && !expr.IsSuppressed) ||
```

Comment:
Why didn't you use `?.` on the second use of expr? Is that just not hittabel ithere?

Reply:
To reach that right part of the `&&`, you need `expr` to be non-null in the left part of the `&&`.

Reply:
Doh!

---
In reply to: [253695325](https://github.com/dotnet/roslyn/pull/32991#discussion_r253695325) [](ancestors = 253695325)